beacon-chain/core/peerdas/p2p_interface.go

```diff
@@ -66,13 +82,49 @@ func VerifyDataColumnsSidecarKZGProofs(sidecars []blocks.RODataColumn) (bool, er
 		}
 	}
 
-	// Verify all the batch at once.
+	// Batch verify that the cells match the corresponding commitments and proofs.
 	verified, err := kzg.VerifyCellKZGProofBatch(commitments, indices, cells, proofs)
 	if err != nil {
-		return false, errors.Wrap(err, "verify cell KZG proof batch")
+		return errors.Wrap(err, "verify cell KZG proof batch")
+	}
+
+	if !verified {
```

Comment:
I know this was not changed in your PR, but do you think its possible to verify the columns, one by one to see which column has the invalid cell proof ? Otherwise this function just returns a generic error

Reply:
Verifying KZG proofs by batch is several order of magnitudes faster than verifying them cells by cells (or even columns by columns since a column could contain at minimum 4 cells.)
For initial sync, it really makes a big difference.

For information, for gossip received columns, this `VerifyDataColumnsSidecarKZGProofs` function is called with a batch of only one column (the received one).

For other cases (by range/root RPC), it is done by batch.

Reply:
Oh no, I meant in the event of failure in batch verification, we try to narrow down which particular column had the invalid cell proof. Or is this expected to be done by the consumer of the method ?

Reply:
OK I see what you mean.
Actually there is IMO two things missing:
1. When, in a batch, a subset (1 or more) of sidecars is faulty according to at least one of the validation criterion, then we currently reject all the batch, even if the majority of sidecars are valid. ==> Waste of resource.
2. When, in a batch, a subset (1 or more) of sidecars is faulty according to at least one of the validation criterion, then we don't know who was the faulty peer, and so no downscore is possible.

==> I added these 2 points into the TODO list, since it's not really related to this PR, and added these tasks would require a new PR just for them.